frontend/components/canvas.tsx

```diff
@@ -1,75 +1,85 @@
-"use client";
+'use client'
 
-import { useDraw } from "@/hooks/useDraw";
-import { useState } from "react";
-import { CompactPicker } from "react-color";
+import { FC, useEffect, useState } from 'react'
+import { useDraw } from '../hooks/useDraw'
+import { ChromePicker } from 'react-color'
 
-export default function Canvas() {
-  const [color, setColor] = useState<string>('#000');
-  const [width, setWidth] = useState<number>(5);
-  const { canvasRef, onMouseDown, clear } = useDraw(drawLine);
+import { io } from 'socket.io-client'
+import { drawLine } from '../util/drawLine'
+const socket = io('http://localhost:3001')
 
-  function drawLine({ prevPoint, currentPoint, context }: Draw) {
-    const { x: currX, y: currY } = currentPoint;
-    const lineColor = color;
-    const lineWidth = width;
+interface pageProps {}
 
-    let startPoint = prevPoint ?? currentPoint;
-    context.beginPath();
-    context.lineWidth = lineWidth;
-    context.strokeStyle = lineColor;
-    context.moveTo(startPoint.x, startPoint.y);
-    context.lineTo(currX, currY);
-    context.stroke();
+type DrawLineProps = {
+  prevPoint: Point | null
+  currentPoint: Point
+  color: string
+}
+
+const page: FC<pageProps> = () => {
+  const [color, setColor] = useState<string>('#000')
+  const { canvasRef, onMouseDown, clear } = useDraw(createLine)
+
+  useEffect(() => {
+    const ctx = canvasRef.current?.getContext('2d')
+
+    socket.emit('client-ready')
+
+    socket.on('get-canvas-state', () => {
+      if (!canvasRef.current?.toDataURL()) return
+      console.log('sending canvas state')
+      socket.emit('canvas-state', canvasRef.current.toDataURL())
+    })
+
+    socket.on('canvas-state-from-server', (state: string) => {
+      console.log('I received the state')
+      const img = new Image()
+      img.src = state
+      img.onload = () => {
+        ctx?.drawImage(img, 0, 0)
+      }
+    })
 
-    context.fillStyle = lineColor;
-    context.beginPath();
-    context.arc(startPoint.x, startPoint.y, 2, 0, 2 * Math.PI);
-    context.fill();
+    socket.on('draw-line', ({ prevPoint, currentPoint, color }: DrawLineProps) => {
+      if (!ctx) return console.log('no ctx here')
+      drawLine({ prevPoint, currentPoint, ctx, color })
+    })
+
+    socket.on('clear', clear)
+
+    return () => {
+      socket.off('draw-line')
+      socket.off('get-canvas-state')
+      socket.off('canvas-state-from-server')
+      socket.off('clear')
+    }
+  }, [canvasRef])
+
+  function createLine({ prevPoint, currentPoint, ctx }: Draw) {
+    socket.emit('draw-line', { prevPoint, currentPoint, color })
```

Comment:
Need to create a new event handler on the backend to accept this "draw-line" event and send it to all the clients in the room, so they can update their canvases

Update `backend\src\socket.ts`, something along the lines of
```
socket.on("draw-line", ( { prevPoint, currentPoint, color }) => {
      const gameId = socket.data.gameId;
      if (gameId !== null) {
        socket.to(gameId).emit("draw-line",  { prevPoint, currentPoint, color });
      }
    });
```

and update `ClientToServerEvents` so our typing expects this event name